File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/lab_01/questao_03/Aquecimento.c
#include <stdio.h>
#include <string.h>

int IsFim(char str[]){
    int result = 1;

    if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M'){
        result = 0;
    }

    return result;
}

int Contar(char str[]){
    int count = 0;
    int n = strlen(str);

    for(int i = 0; i < n; i++){
        if(str[i] >= 'A' && str[i] <= 'Z'){
            count++;
        }
    }

    return count;
}

int main(void){
    char line[100];

    scanf("%[^\n]", line);
    while(IsFim(line) != 0){
        printf("%d\n", Contar(line));

        scanf(" %[^\n]", line);
    }

  return 0;
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/lab_01/questao_01/Aquecimento.java
import java.util.*;

class Aquecimento{
    public static int IsFim(String str){
        int result = 0;

        if(str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'){
            result = 1;
        }

        return result;
    }

    public static int Contar(String str){
        int count = 0;

        for(int i = 0; i < str.length(); i++){
            if(str.charAt(i) >= 'A' && str.charAt(i) <= 'Z'){
                count++;
            }
        }

        return count;
    }

    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        String line;

        line = sc.nextLine();
        while(IsFim(line) != 1){
            System.out.println(Contar(line) + " ");

            line = sc.nextLine();
        }

        sc.close();
    }
}